lib/compat.php

```diff
@@ -441,18 +441,17 @@ function gutenberg_inject_default_block_context( $args ) {
 			$registry = WP_Block_Type_Registry::get_instance();
 			$block_type = $registry->get_registered( $block->name );
 
-
 			// For WordPress versions that don't support the context API.
 			if ( ! $block->context ) {
 				$block->context = array();
 			}
 
 			// Inject the post context if not done by Core.
-			$needs_post_id = empty( $block_type->uses_context ) && in_array( 'postId', $block_type->uses_context, true );
+			$needs_post_id = false;
```

Comment:
This is explicitly stating what the current code does. Any array is never both empty and has this key. A proper fix seems to cause e2e test failures. See 5076c4d0ab5bf8ac563e7b368e88ee9778b82dd2.